ibm/config.go

```diff
@@ -594,91 +594,368 @@ func (sess clientSession) IAMNamespaceAPI() (*ns.IbmCloudFunctionsNamespaceAPIV1
 
 // CIS Zones Service
 func (sess clientSession) CisZonesV1ClientSession() (*ciszonesv1.ZonesV1, error) {
+	if sess.cisZonesErr == nil {
+		cisZonesV1Opt := &ciszonesv1.ZonesV1Options{
+			URL:           sess.cisZonesV1Client.Service.Options.URL,
+			Crn:           core.StringPtr(""),
+			Authenticator: sess.cisZonesV1Client.Service.Options.Authenticator,
+		}
+		cisZonesClient, cisZonesErr := ciszonesv1.NewZonesV1(cisZonesV1Opt)
+		if cisZonesErr != nil {
+			return nil, fmt.Errorf(
+				"Error occured while configuring CIS Zones client: %s",
+				cisZonesErr)
+		}
+		return cisZonesClient, nil
+	}
 	return sess.cisZonesV1Client, sess.cisZonesErr
 }
 
 // CIS DNS Service
 func (sess clientSession) CisDNSRecordClientSession() (*cisdnsrecordsv1.DnsRecordsV1, error) {
+	if sess.cisDNSErr == nil {
+		cisDNSRecordsOpt := &cisdnsrecordsv1.DnsRecordsV1Options{
+			URL:            sess.cisDNSRecordsClient.Service.Options.URL,
+			Crn:            core.StringPtr(""),
+			ZoneIdentifier: core.StringPtr(""),
+			Authenticator:  sess.cisDNSRecordsClient.Service.Options.Authenticator,
+		}
+		cisDNSRecordsClient, cisDNSErr := cisdnsrecordsv1.NewDnsRecordsV1(cisDNSRecordsOpt)
+		if cisDNSErr != nil {
+			return nil, fmt.Errorf(
+				"Error occured while configuring CIS DNS client: %s", cisDNSErr)
+		}
+		return cisDNSRecordsClient, nil
+	}
 	return sess.cisDNSRecordsClient, sess.cisDNSErr
 }
 
 // CIS GLB Pool
 func (sess clientSession) CisGLBPoolClientSession() (*cisglbpoolv0.GlobalLoadBalancerPoolsV0, error) {
+	if sess.cisGLBPoolErr == nil {
```

Comment:
Same here:

```go
if sess.cisGLBPoolErr != nil {
	return sess.cisGLBPoolClient, sess.cisGLBPoolErr
}

cisGLBPoolOpt := &cisglbpoolv0.GlobalLoadBalancerPoolsV0Options{
	URL:           sess.cisGLBPoolClient.Service.Options.URL,
	Crn:           core.StringPtr(""),
	Authenticator: sess.cisGLBPoolClient.Service.Options.Authenticator,
}
cisGLBPoolClient, cisGLBPoolErr :=
	cisglbpoolv0.NewGlobalLoadBalancerPoolsV0(cisGLBPoolOpt)
if cisGLBPoolErr != nil {
	return nil, fmt.Errorf(
		"Error occured while configuring CIS GLB Pool client: %s",
		cisGLBPoolErr)
}

return cisGLBPoolClient, nil
```

Reply:
I have upgraded sdk and regenerated cis sdk. Modified the code as per your suggession.